src/tiled/mapobjectmodel.cpp

```diff
@@ -242,8 +253,11 @@ QVariant MapObjectModel::headerData(int section, Qt::Orientation orientation,
 {
     if (role == Qt::DisplayRole && orientation == Qt::Horizontal) {
         switch (section) {
-        case 0: return tr("Name");
-        case 1: return tr("Type");
+        case Name: return tr("Name");
+        case Type: return tr("Type");
+        case Id: return tr("Id");
```

Comment:
I've used "ID" in the Properties view for this field, so this should also use uppercase.